static/js/components/FollowupRequestForm.jsx

```diff
@@ -0,0 +1,154 @@
+import React, { useState, useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import PropTypes from "prop-types";
+import Select from "@material-ui/core/Select";
+import InputLabel from "@material-ui/core/InputLabel";
+import MenuItem from "@material-ui/core/MenuItem";
+import Form from "@rjsf/material-ui";
+import * as Actions from "../ducks/source";
+
+import "react-datepicker/dist/react-datepicker-cssmodules.css";
+
+const FollowupRequestForm = ({
+  obj_id,
+  instrumentList,
+  instrumentFormParams,
+}) => {
+  const dispatch = useDispatch();
+  const telescopeList = useSelector((state) => state.telescopes.telescopeList);
+  const { allocationList } = useSelector((state) => state.allocations);
+  const allGroups = useSelector((state) => state.groups.all);
+  const [selectedInstrumentId, setSelectedInstrumentId] = useState(null);
+  const [selectedAllocationId, setSelectedAllocationId] = useState(null);
+  const allocationInstIds = allocationList.map((a) => a.instrument_id);
+  const instIds = Object.keys(instrumentFormParams)
+    .map((k) => parseInt(k, 10))
+    .filter((i) => allocationInstIds.includes(i));
+
+  // initialize the form fields that are outside of the schema
+  // this needs to be in a useEffect hook so that the state setters are not
+  // called unconditionally, resulting in an infinite re-render loop
+  useEffect(() => {
+    const defaultInstId = instIds[0];
+    const defaultAllocationId = allocationList.filter(
+      (allocation) => allocation.instrument_id === defaultInstId
+    )[0]?.id;
+    setSelectedInstrumentId(defaultInstId);
+    setSelectedAllocationId(defaultAllocationId);
+  }, [allocationList, instrumentFormParams]);
+
+  if (instIds.length === 0) {
+    return <h3>No robotic instruments available...</h3>;
+  }
+
+  if (allGroups.length === 0) {
+    return <h3>Loading...</h3>;
+  }
+
+  if (telescopeList.length === 0) {
+    return <h3>Loading...</h3>;
+  }
+
+  const groupLookUp = allGroups.reduce((r, a) => {
+    r[a.id] = a;
+    return r;
+  });
+
+  const telLookUp = {};
+  telescopeList.forEach((tel) => {
+    telLookUp[tel.id] = tel;
+  });
+
+  const instIDToName = {};
+  const instLookUp = {};
+  instrumentList.forEach((instrumentObj) => {
+    instIDToName[instrumentObj.id] = instrumentObj.name;
+    instLookUp[instrumentObj.id] = instrumentObj;
+  });
+
+  const handleSelectedInstrumentChange = (e) => {
+    setSelectedInstrumentId(e.target.value);
+  };
+
+  const handleSelectedAllocationChange = (e) => {
+    setSelectedAllocationId(e.target.value);
+  };
+
+  const handleSubmit = ({ formData }) => {
+    const json = {
+      obj_id,
+      allocation_id: selectedAllocationId,
+      payload: formData,
+    };
+    dispatch(Actions.submitFollowupRequest(json));
+  };
```

Comment:
Should await response and reset if successful?

Reply:
Unless we expect WS message to trigger auto-reset/refresh

Reply:
yes the WS handles this.